src/ds/cli_helper.h

```diff
@@ -58,4 +63,61 @@ namespace cli
     return option;
   }
 
+  struct ParsedMemberInfo
+  {
+    std::string cert_file;
+    std::string keyshare_pub_file;
+
+    ParsedMemberInfo(const std::string& cert, const std::string& keyshare_pub) :
+      cert_file(cert),
+      keyshare_pub_file(keyshare_pub)
+    {}
+  };
+
+  CLI::Option* add_member_info_option(
+    CLI::App& app,
+    std::vector<ParsedMemberInfo>& parsed,
+    const std::string& option_name,
+    const std::string& option_desc)
+  {
+    CLI::callback_t fun = [&option_name, &parsed](CLI::results_t res) {
+      parsed.clear();
+      for (const auto& r : res)
+      {
+        auto found = r.find_last_of(",");
+        if (found == std::string::npos)
+        {
+          throw CLI::ValidationError(
+            option_name,
+            "Member info is not in format "
+            "member_cert.pem,member_encryption_public_key.pem");
+        }
+
+        auto cert = r.substr(0, found);
+        auto keyshare_pub = r.substr(found + 1);
+
+        // Validate that member certificate and public encryption key exist
+        auto validator = CLI::detail::ExistingFileValidator();
+        auto err_str = validator(cert);
+        if (!err_str.empty())
+        {
+          throw CLI::ValidationError(option_name, err_str);
+        }
+        err_str = validator(keyshare_pub);
+        if (!err_str.empty())
+        {
+          throw CLI::ValidationError(option_name, err_str);
+        }
+
+        parsed.emplace_back(cert, keyshare_pub);
+      }
+      return true;
+    };
+
+    auto* option = app.add_option(option_name, fun, option_desc, true);
+    option->type_name("member_cert.pem,member_kshare_pub.pem.pem")
+      ->type_size(-1);
```

Comment:
FYI, `type_size(-1)` tells CLI11 that the option is a vector.

Reply:
.pem.pem?